src/client/eppo-client.ts

```diff
@@ -76,54 +78,82 @@ export interface IContainerExperiment<T> {
 }
 
 export default class EppoClient {
-  private readonly queuedAssignmentEvents: IAssignmentEvent[] = [];
-  private assignmentLogger?: IAssignmentLogger;
-  private readonly queuedBanditEvents: IBanditEvent[] = [];
+  private readonly eventQueue: NamedEventQueue<unknown>;
+  private readonly assignmentEventsQueue: BoundedEventQueue<IAssignmentEvent> =
+    newBoundedArrayEventQueue<IAssignmentEvent>('assignments');
+  private readonly banditEventsQueue: BoundedEventQueue<IBanditEvent> =
+    newBoundedArrayEventQueue<IBanditEvent>('bandit');
+  private readonly banditEvaluator = new BanditEvaluator();
   private banditLogger?: IBanditLogger;
-  private isGracefulFailureMode = true;
-  private assignmentCache?: AssignmentCache;
   private banditAssignmentCache?: AssignmentCache;
+  private configurationRequestParameters?: FlagConfigurationRequestParameters;
+  private banditModelConfigurationStore?: IConfigurationStore<BanditParameters>;
+  private banditVariationConfigurationStore?: IConfigurationStore<BanditVariation[]>;
+  private flagConfigurationStore: IConfigurationStore<Flag | ObfuscatedFlag>;
+  private assignmentLogger?: IAssignmentLogger;
+  private assignmentCache?: AssignmentCache;
+  // whether to suppress any errors and return default values instead
+  private isGracefulFailureMode = true;
+  private isObfuscated: boolean;
   private requestPoller?: IPoller;
   private readonly evaluator = new Evaluator();
-  private readonly banditEvaluator = new BanditEvaluator();
 
-  constructor(
-    private flagConfigurationStore: IConfigurationStore<Flag | ObfuscatedFlag>,
-    private banditVariationConfigurationStore?: IConfigurationStore<BanditVariation[]>,
-    private banditModelConfigurationStore?: IConfigurationStore<BanditParameters>,
-    private configurationRequestParameters?: FlagConfigurationRequestParameters,
-    private isObfuscated = false,
-  ) {}
+  constructor({
+    eventQueue = new ArrayBackedNamedEventQueue('events'),
```

Comment:
thanks for the named parameters.

🤔 I do NOT consider this a breaking change as users are not using this constructor directly; they are going through `init`. Am I misunderstanding?

Reply:
I think that's correct sir!

Reply:
According to strict semver, this does require a major version bump, however, as you pointed out, since it's only used internally, it doesn't affect end user